database/seeders/PlayerAchievementsSeeder.php

```diff
@@ -0,0 +1,154 @@
+                        'unlocked_hardcore_at' => $hardcore ? $date : null,
+                    ]);
+
+                    // time advances
+                    $date = $date->addSeconds(random_int(0, random_int(10, 500) + random_int(10, 500) + random_int(10, 500) + random_int(10, 500)));
+
+                    if (random_int(0, $achievementsRemaining) === 0) {
+                        // player gives up
+                        break;
+                    }
+
+                    if ($hardcore && random_int(1, 40) === 1) {
+                        // player switches to softcore
+                        $hardcore = false;
+                    }
+
+                    if (random_int(0, $keepPlayingChance) === 0) {
+                        // player gave up
+                        break;
+                    }
+                    $keepPlayingChance = max(0, $keepPlayingChance - random_int(0, (int) floor($achievementsRemaining / 3)));
+
+                    if (random_int(0, (int) floor($numAchievements / $num_sessions)) > $achievementsRemaining) {
+                        // player takes a break
+
+                        $playerSession->rich_presence = ucfirst($faker->words(random_int(2, 10), true));
+                        $playerSession->rich_presence_updated_at = $date;
+                        $playerSession->duration = $date->diffInMinutes($playerSession->created_at);
+                        $playerSession->save();
+
+                        $date = $date->addMinutes((int) sqrt(random_int(500 * 500, 100000 * 100000))); // 8 hours to three month break, weighted toward shorter period
+                        if ($date < Carbon::now()) {
+                            $playerSession = $resumePlayerSessionAction->execute($user, $game, timestamp: $date);
+                            $playerSession->created_at = $date;
+                            $date = $date->addSeconds(random_int(200, 2000));
+                        }
+                    }
+
+                    if ($date > Carbon::now()) {
+                        break;
+                    }
+                }
+
+                // finalize the session
+                $playerSession->rich_presence = ucfirst($faker->words(random_int(2, 10), true));
+                $playerSession->rich_presence_updated_at = $date;
+                $playerSession->duration = $date->diffInMinutes($playerSession->created_at);
+                $playerSession->save();
+
+                // aggregate metrics for player
+                $updatePlayerGameMetricsAction->execute($playerGame, silent: true);
+
+                // update points
+                $playerGame->refresh();
+                $user->RAPoints += $playerGame->points_hardcore;
+                $user->RASoftcorePoints += $playerGame->points;
+                $user->save();
+            }
+
+            // update player count and unlock metrics
+            (new UpdateGameMetricsAction())->execute($game);
```

Comment:
this updates the player count for the game and the unlock metrics for each achievement in the game